### PR TITLE
ncm-freeipa: initial support

### PR DESCRIPTION
**Pull request by stdweird**
ncm-freeipa adds support for configuring (parts of) FreeIPA (http://www.freeipa.org) server and client

### CODE REVIEW

**Comment by stdweird** [comment]
Has lots of dependecies on other PRs. Not adding it to 16.4 milestone, enough PRs open already :smile:

File: ncm-freeipa/src/main/pan/components/freeipa/schema.pan [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    @{first name}
+    'givenname' : string
+    @{group name (must be a configured group to retrieve the gid)}
+    'group' ? string with exists('/software/components/${project.artifactId}/server/groups/'+SELF)
+    @{homedirectory}
+    'homedirectory' ? string
+    @{gecos}
+    'gecos' ? string
+    @{loginshell}
+    'loginshell' ? string with match(SELF, '^/bin/(ba|c)sh$')
```
Comment (ned21) [inline ncm-freeipa/src/main/pan/components/freeipa/schema.pan]: The regex seems a bit restrictive. /etc/shells on my box is:
```
/bin/bash
/bin/csh
/bin/ksh
/bin/sh
/bin/tcsh
/bin/zsh
```
which makes for a pretty horrible regex!

Comment (jrha) [inline ncm-freeipa/src/main/pan/components/freeipa/schema.pan]: `^/bin/(ba|tc|[ckz])?sh$`
or perhaps just...
`^/bin/\w+sh$`

**Comment by stdweird** [comment]
Test this please

**Comment by stdweird** [comment]
restest this please

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
retest this please

**Pull request merged**